keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+     * This operation requires the secrets/recover permission.
+     *
+     * @param name The name of the deleted secret to be recovered.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the {@link Secret} that was recovered
+     */
+    public Mono<RestResponse<Secret>> recoverDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.recoverDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Backs up the specified secret.
+     * Requests that a backup of the specified secret be downloaded to the client. All versions of the secret will be downloaded.
+     * This operation requires the secrets/backup permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the {@link SecretBackup} backup of the secret.
+     */
+    public Mono<RestResponse<SecretBackup>> backupSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.backupSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Restores a backed up secret to a vault.
+     * Restores a backed up secret, and all its versions, to a vault.
+     * This operation requires the secrets/restore permission.
+     *
+     * @param backup The backup blob associated with the secret.
+     * @throws NullPointerException if {@code SecretBackup} is {@code null}.
+     * @return the {@link Secret} that was restored.
+     */
+    public Mono<RestResponse<Secret>> restoreSecretAsync(SecretBackup backup) {
+        Objects.requireNonNull(backup, "The Secret backup parameter cannot be null.");
+        Objects.requireNonNull(backup.value(), "The backup value cannot be null.");
+        SecretRestoreRequestOptions parameters = new SecretRestoreRequestOptions()
+                                                .withSecretBackup(backup.value());
+        return service.restoreSecret(vaultEndPoint, API_VERSION, ACCEPT_LANGUAGE, parameters);
+    }
+
+
+    /**
+     * List {@link Secret} secrets in a specified key vault.
+     * The Get Secrets operation is applicable to the entire vault. However, only the base secret identifier and its attributes are provided in the response.
+     * Individual secret versions are not listed in the response. This operation requires the secrets/list permission.
+     *
+     * @return A Flux of Secrets. The Flux contains all of the secrets in the vault.
+     */
+    public Flux<SecretInfo> listSecretsAsync() {
+        Mono<RestResponse<Page<SecretInfo>>> result = service.getSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE);
+
+        return getPagedSecrets(result);
+    }
+
+    /**
+     * Lists {@link DeletedSecret} secrets of the key vault.
```

Comment:
Same pattern here (and everywhere) `{@link DeletedSecret deleted secrets} of the key vault`

Reply:
updated.